crates/ae-plugin/src/lib.rs

```diff
@@ -901,17 +901,23 @@ impl Plugin {
                 };
 
                 let effect_settings = self.apply_settings(params)?;
-                let res: Result<(), Box<dyn std::error::Error>> = (|| {
-                    let mut destination = File::create(preset_path).map_err(Box::new)?;
-                    self.settings
-                        .write_json_to_io(&effect_settings, &mut destination)
-                        .map_err(Box::new)?;
-
-                    Ok(())
-                })();
-                if let Err(e) = res {
-                    out_data.set_error_msg(&format!("Error saving preset: {}", e.kind()));
-                    return Ok(());
+                let mut destination = match File::create(preset_path).map_err(Box::new) {
+                    Ok(res) => res,
+                    Err(e) => {
+                        out_data.set_error_msg(&format!("Error saving preset: {}", e.kind()));
+                        return Ok(());
+                    }
+                };
+
+                match self
+                    .settings
+                    .write_json_to_io(&effect_settings, &mut destination)
+                {
+                    Ok(_) => {}
+                    Err(e) => {
+                        out_data.set_error_msg(&format!("Error saving preset: {:?}", e));
+                        return Ok(());
+                    }
```

Comment:
I *think* the error here is just that `e.kind` doesn't exist on `Box<dyn std::error::Error>`, but that's OK because we can just `Display` it instead. I can make this change myself.

```suggestion
                let res: Result<(), Box<dyn std::error::Error>> = (|| {
                    let mut destination = File::create(preset_path).map_err(Box::new)?;
                    self.settings
                        .write_json_to_io(&effect_settings, &mut destination)
                        .map_err(Box::new)?;

                    Ok(())
                })();
                if let Err(e) = res {
                    out_data.set_error_msg(&format!("Error saving preset: {}", e));
                    return Ok(());
```